static/cypress/integration/main.spec.ts

```diff
@@ -26,52 +26,107 @@ function intercept_next_image_with_failure() {
   ).as("get_next_image");
 }
 
+function intercept_login() {
+  cy.intercept("POST", `${Cypress.env("API_URL")}/login`, {
+    statusCode: 204,
+  }).as("login");
+}
+
+function intercept_login_with_failure() {
+  cy.intercept("POST", `${Cypress.env("API_URL")}/login`, {
+    statusCode: 401,
+  }).as("login");
+}
+
+function login() {
+  intercept_login();
+  return cy
+    .visit("/")
+    .get("input[name='username']")
+    .type("AnnotoUser#1337")
+    .get("input[name='password']")
+    .type("test1234")
+    .get("button#submit")
+    .click()
+    .wait("@login");
+}
+
 describe("Application", () => {
   it("visits the app root url", () => {
-    cy.visit("/");
-    cy.contains("h1", "Annoto");
+    cy.visit("/").contains("h1", "Annoto");
+  });
+});
+
+describe("LoginView", () => {
+  it("logs the user in when valid login data is provided", () => {
+    intercept_login();
+    intercept_get_task();
+    intercept_next_image();
+    cy.visit("/")
+      .get("input[name='username']")
+      .type("AnnotoUser#1337")
+      .get("input[name='password']")
+      .type("test1234")
+      .get("button#submit")
+      .click()
+      .wait("@login")
+      .get("#task-view");
```

Comment:
Cant you use your login function here too? And below for the wrong password, maybe paramaterize the login function to take the password and username to be logged in.

Reply:
Maybe I'm wrong, but I think I too realized that, but only in #80 
But even there, I'm not completely happy yet with code duplications - But I'd suggest to have a look at it there.
The problem about taking username/password as a parameter is that they will not be validated at all - there is a fixed mocked response. And whereas we need the `intercept_login` here, below we need `intercept_login_with_failure`. I mean, we could validate the given username/password in a new `intercept_login` method that then either intercepts the request with success or failure, depending on username and password - but that would kinda "re-implement" the server logic and thus be rather pointless (and against the idea of intercepting) in my mind :/

Reply:
Ok i see, yeah lets see how we do that in #80 then 👍